m4/slepc.m4

```diff
@@ -16,46 +16,80 @@ AC_DEFUN([CONFIGURE_SLEPC],
-                ],
+                  AC_MSG_RESULT([<<< Could not infer SLEPC_DIR from PETSC_DIR, SLEPc disabled. >>>])
+                ])
+
+          dnl Check for different flavors of PETSc installs.
+          AS_IF([test "x$SLEPC_DIR" != "x"],
                 [
+                  dnl 1.) A standalone (make install) SLEPc build
+                  dnl 2.) SLEPc in a PETSC_ARCH directory
+                  dnl 3.) SLEPc in /usr/lib/slepc
+                  slepc_standalone=no
+                  slepc_in_petsc_arch=no
+                  slepc_in_usr_lib=no
+
+                  dnl 1.)
                   AC_CHECK_HEADER([$SLEPC_DIR/include/slepcversion.h],
-                                  [SLEPC_INCLUDE="-I$SLEPC_DIR/include"],
-                                  [
-                                    AC_MSG_RESULT(<<< Invalid "\$SLEPC_DIR" detected (slepcversion.h not found). SLEPc disabled. >>>)
-                                    unset SLEPC_DIR
-                                    enableslepc=no
-                                  ])
+                                  [SLEPC_INCLUDE="-I$SLEPC_DIR/include"
+                                   slepc_standalone=yes],[])
+
+                  dnl 2.)
+                  AS_IF([test "x$slepc_standalone" != "xyes" && test "x$PETSC_ARCH" != "x"],
+                        [
+                          AC_CHECK_HEADER([$SLEPC_DIR/$PETSC_ARCH/include/slepcversion.h],
+                                          [SLEPC_INCLUDE="-I$SLEPC_DIR/$PETSC_ARCH/include"
+                                          slepc_in_petsc_arch=yes],[])
+                        ])
+
+                  dnl 3.)
+                  AS_IF([test "x$slepc_standalone" != "xyes" && test "x$slepc_in_petsc_arch" != "xyes"],
+                        [
+                          AC_CHECK_HEADER([/usr/lib/slepc/slepcversion.h],
+                                          [
+                                            export SLEPC_DIR=/usr/lib/slepc
+                                            SLEPC_INCLUDE="-I$SLEPC_DIR/include"
+                                          ],
+                                          [
+                                            AC_MSG_RESULT([<<< No valid SLEPC installation found, SLEPc disabled. >>>])
+                                            unset SLEPC_DIR
+                                            enableslepc=no
+                                          ])
+                        ])
                 ])
-
-            dnl I didn't check this code branch since I don't use a
-            dnl PETSC_ARCH, but running AC_CHECK_HEADER on slepcconf.h
-            dnl should be pretty safe... from what I can tell it just
-            dnl #defines a few things.
-            AS_IF([test "x$PETSC_ARCH" != "x"],
-                  [AC_CHECK_HEADER([$SLEPC_DIR/$PETSC_ARCH/include/slepcconf.h],
-                                   [SLEPC_INCLUDE="$SLEPC_INCLUDE -I$SLEPC_DIR/$PETSC_ARCH/include"])])
```

Comment:
Okay, this deletion looks like the problem.  Without it we never notice that slepcconf.h is under the $PETSC_ARCH subdirectory, we never add that entry to $SLEPC_INCLUDE, and the later build fails.